### PR TITLE
Develop to master

### PR DESCRIPTION
**Pull request by ThrawnCA**
- Prepare for Python 3
- Make Flake8 happy
- Don't force the version of `ckanext-report`, just ensure it's present
- Add Github Actions workflow to replace TravisCI

### CODE REVIEW

File: ckanext/archiver/tasks.py [hunk 1]
```diff
@@ -356,9 +357,10 @@ def _save(status_id, exception, resource, url_redirected_to=None,
         'site_url': config.get('ckan.site_url_internally') or config['ckan.site_url'],
         'cache_url_root': config.get('ckanext-archiver.cache_url_root'),
         'previous': Archival.get_for_resource(resource_id)
-        }
+    }
     try:
         download_result = download(context, resource)
+        e = {'args': ''}
```
Comment (duttonw) [inline ckanext/archiver/tasks.py:363]: why is this needed?

**Review by duttonw — APPROVED** [review]

File: ckanext/archiver/tasks.py (hunk 1 above)
Comment (ThrawnCA) [reply ckanext/archiver/tasks.py:363]: I think it's possible to have a non-OK status code without an actual exception, which would result in an error at line 383.
